apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts

```diff
@@ -112,24 +118,26 @@ export class SubpoenaService {
     user: User,
   ): Promise<DeliverResponse> {
     try {
-      const subpoena = await this.createSubpoena(defendant)
-
-      const createdSubpoena = await this.policeService.createSubpoena(
-        theCase,
-        defendant,
-        subpoenaFile,
-        user,
-      )
-
-      if (!createdSubpoena) {
-        this.logger.error('Failed to create subpoena file for police')
-        return { delivered: false }
-      }
-
-      await this.subpoenaModel.update(
-        { subpoenaId: createdSubpoena.subpoenaId },
-        { where: { id: subpoena.id } },
-      )
+      await this.sequelize.transaction(async (transaction) => {
+        const subpoena = await this.createSubpoena(defendant, transaction)
+
+        const createdSubpoena = await this.policeService.createSubpoena(
+          theCase,
+          defendant,
+          subpoenaFile,
+          user,
+        )
+
+        if (!createdSubpoena) {
+          this.logger.error('Failed to create subpoena file for police')
+          return { delivered: false }
+        }
```

Comment:
I am not absolutely sure, but I don't think this will rollback the transaction. I think you need to throw an exception for that.

Reply:
I checked the documentation and my understanding is correct. You need to throw to trigger a rollback. The catch clause below returns the right delivered status so that should be fine.

Reply:
Ok weird because I tried it locally and it rolled back! I'll try again tomorrow to confirm but I can always add in an exception just in case

Reply:
Hér er skjölunin fyrir þetta:

   * Start a transaction. When using transactions, you should pass the transaction in the options argument
   * in order for the query to happen under that transaction
   *
   * ```js
   *   try {
   *     const transaction = await sequelize.transaction();
   *     const user = await User.findOne(..., { transaction });
   *     await user.update(..., { transaction });
   *     await transaction.commit();
   *   } catch(err) {
   *     await transaction.rollback();
   *   }
   * })
   * ```
   *
   * A syntax for automatically committing or rolling back based on the promise chain resolution is also
   * supported:
   *
   * ```js
   * try {
   *   await sequelize.transaction(transaction => { // Note that we pass a callback rather than awaiting the call with no arguments
   *     const user = await User.findOne(..., {transaction});
   *     await user.update(..., {transaction});
   *   });
   *   // Committed
   * } catch(err) {
   *   // Rolled back
   *   console.error(err);
   * }
   * ```

Þú getur annað hvort búið til transaction og séð sjálf um commit og rollback, eða sent callback inn í transaction fallið og látið transaction fallið sjá um commit og rollback. Til að callback fallið geti rollbackað er það væntanlega með try/catch utan um callbackið og þess vegna þarf throw.

Reply:
Aa ok ég veit af hverju þetta virkaði þegar ég prófaði locally - af því þá var createsubpoena kallið að kasta exception ekki skila villu eða tómu svari þannig ég datt ekkert inn í if(!created)

Reply:
Í rauninni gæti ég bara tekið þessa if setningu alveg út, er að kasta exceptions ef eitthvað klikkar í lögreglukallinu þannig þetta er eiginlega unreachable kóði